File: src/backend/src/routes/logout.ts
import { generateResponse } from "../../../frontend/src/shared/ServerResponse";
import { Status } from "../../../frontend/src/shared/Status";

export function handleLogout(req: any, res: any) {
  req.session.destroy((err: any) => {
    if (err) {
      console.error("Fehler beim Löschen der Session:", err);
      return generateResponse(
        res,
        Status.ERROR,
        "An error occured while logging out.",
      );
    }
    res.clearCookie("connect.sid");
    return generateResponse(res, Status.OK, "");
  });
}

File: src/backend/src/routes/logout.py
from fastapi import APIRouter, Request
from src.utils.status import Status
from src.utils.server_response import generate_response

router = APIRouter()

@router.get("/logout")
async def handle_logout(request: Request):
    request.session.clear() 
    response = generate_response(Status.OK, "")
    response.delete_cookie("session") # Standard-Cookie von Starlette löschen
    return response